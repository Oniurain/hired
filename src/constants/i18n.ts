import { get } from 'lodash'

import { isDev } from '../lib/env'
import * as log from '../lib/log'

const translationMap = Object.freeze({
  entryList: {
    empty: 'You do not have entries so far. Hit the big green button!'
  },
  projectPage: {
    empty: 'You have no projects yet. Use the "+" to configure some.'
  }
})

const translate = (key: string): string  => {
  const translation = get(translationMap, key, null)

  if (translation === null) {
    log.error('[i18n]: Could not find translation for key', key)
  }

  return translation || (isDev() ? `[MISSING TRANSLATION FOR KEY ${key}]` : '')
}

export {
  translate
}
