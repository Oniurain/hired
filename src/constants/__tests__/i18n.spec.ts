import { translate } from '../i18n'

describe('i18n', () => {
  it('should return the correct string for an existing key', () => {
    expect(translate('entryList.empty')).toMatchSnapshot()
  })

  it('should return empty string for an unknown key', () => {
    expect(translate('foo.bar.buzz')).toBe('')
  })
})
