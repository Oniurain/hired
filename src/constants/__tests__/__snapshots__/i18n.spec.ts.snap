// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n should return the correct string for an existing key 1`] = `"You do not have entries so far. Hit the big green button!"`;
